### PR TITLE
Update taxonomy.api.php

### LINKED ISSUE #4635: Add documentation for `hook_taxonomy_term_page_alter()`.
[Line 69](https://github.com/backdrop/backdrop/blob/1.x/core/modules/taxonomy/taxonomy.pages.inc#L69) of taxonomy.pages.inc has this:

> `backdrop_alter('taxonomy_term_page', $build);`

This means that you can implement `hook_taxonomy_term_page_alter()`, however that hook isn't documented. It should be detailed in [taxonomy.api.php](https://github.com/backdrop/backdrop/blob/1.x/core/modules/taxonomy/taxonomy.api.php).

Todo: 

* add an example use of `hook_taxonomy_term_page_alter()` into `taxonomy.api.php`. Here is an example I use in the wild:

```php
/**
 * Implements hook_taxonomy_term_page_alter().
 */
function custom_taxonomy_term_page_alter(&$build) {
  // Remove the list of nodes with this term.
  unset($build['nodes']);
  // Remove the pager for the list of nodes.
  unset($build['pager']);
}

```

### PR DESCRIPTION
**Pull request by remote007**
hook_taxonomy_term_page_alter() isn't documented.
Documented the required example. Fixes: https://github.com/backdrop/backdrop-issues/issues/4635

### CODE REVIEW

**Review by remote007 — COMMENTED** [review]
Removed extra comment.

**Comment by laryn** [comment]
@PadminiRai Thanks for this! 

It should be cleaned up a bit for coding standards yet. See this documentation:

 - https://docs.backdropcms.org/doc-standards#functions

Some suggestions:

 - add a blank line between the previous function and the docblock
 - the summary line should be added at the top of the docblock (see the function above for an example) -- in this case something like `Alter the results of taxonomy_term_page().`
 - add spaces on each line between the `*` and the text in the docblock
 - (this may be personal preference) the original function passes the variable as `$build` so it might be cleaner to use `&$build` instead of `&$var_list`
 - remove your name and add `@param` instead, and add a description of the variable on the next line, such as `A renderable array representing the node listing for a taxonomy term.`
 - add `@see taxonomy_term_page()` at the end of the docblock (to reference the original function that this hook alters)

I think we may be able to clarify the description text that goes along with it as well. Here's what we have currently:

>  This hook is called to remove the list of nodes with a renderable array representing the taxonomy term content passed as an argument and pager for the list of nodes.

Maybe something like this? Others may have suggestions, too.

> This hook is called after the node listing for a taxonomy term page has been assembled in a structured array and may be used for altering that listing.

**Comment by remote007** [comment]
> @PadminiRai Thanks for this!
> 
> It should be cleaned up a bit for coding standards yet. See this documentation:
> 
> * https://docs.backdropcms.org/doc-standards#functions
> 
> Some suggestions:
> 
> * add a blank line between the previous function and the docblock
> * the summary line should be added at the top of the docblock (see the function above for an example) -- in this case something like `Alter the results of taxonomy_term_page().`
> * add spaces on each line between the `*` and the text in the docblock
> * (this may be personal preference) the original function passes the variable as `$build` so it might be cleaner to use `&$build` instead of `&$var_list`
> * remove your name and add `@param` instead, and add a description of the variable on the next line, such as `A renderable array representing the node listing for a taxonomy term.`
> * add `@see taxonomy_term_page()` at the end of the docblock (to reference the original function that this hook alters)
> 
> I think we may be able to clarify the description text that goes along with it as well. Here's what we have currently:
> 
> > This hook is called to remove the list of nodes with a renderable array representing the taxonomy term content passed as an argument and pager for the list of nodes.
> 
> Maybe something like this? Others may have suggestions, too.
> 
> > This hook is called after the node listing for a taxonomy term page has been assembled in a structured array and may be used for altering that listing.

Updated the PR.

File: core/modules/taxonomy/taxonomy.api.php [hunk 1]
```diff
@@ -272,6 +272,23 @@ function hook_taxonomy_term_view_alter(&$build) {
   $build['#post_render'][] = 'my_module_taxonomy_term_post_render';
 }
 
+/**
+ * Alter the results of taxonomy_term_page()
+ *
+ * This hook is called after the node listing for a taxonomy term page has been assembled in a structured array
+ * and may be used for altering that listing.
```
Comment (klonos) [inline core/modules/taxonomy/taxonomy.api.php]: We should stick to the rule of 80 characters max per line:

```suggestion
 * This hook is called after the node listing for a taxonomy term page has
 * been assembled in a structured array and may be used for altering that
 * listing.
```

See https://docs.backdropcms.org/doc-standards#general

> Lines containing comments (including docblocks) must wrap as close to 80 characters as possible without going over, ...

File: core/modules/taxonomy/taxonomy.api.php [hunk 2]
```diff
@@ -272,6 +272,23 @@ function hook_taxonomy_term_view_alter(&$build) {
...
+ *
+ * @param $build
+ *   A renderable array representing the taxonomy term content.
+ *
+ * @see taxonomy_term_page()
+ */
+function hook_taxonomy_term_page_alter(&$build) {
+  // Remove the list of nodes with this term (build).
+  unset($build['nodes']);
+  // To remove the pager for the list of nodes.
```
Comment (klonos) [inline core/modules/taxonomy/taxonomy.api.php]:
```suggestion
  // Remove the pager for the list of nodes.
```

**Review by remote007 — COMMENTED** [review]
Updated according to the proposed changes.

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

File: core/modules/taxonomy/taxonomy.api.php [hunk 3]
```diff
@@ -272,6 +272,24 @@ function hook_taxonomy_term_view_alter(&$build) {
   $build['#post_render'][] = 'my_module_taxonomy_term_post_render';
 }
 
+/**
+ * Alter the results of taxonomy_term_page()
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.api.php]: Need a full-stop here (https://docs.backdropcms.org/doc-standards#general):

```suggestion
 * Alter the results of taxonomy_term_page().
```

File: core/modules/taxonomy/taxonomy.api.php [hunk 4]
```diff
@@ -272,6 +272,24 @@ function hook_taxonomy_term_view_alter(&$build) {
...
+ *
+ * This hook is called after the node listing for a taxonomy term page has
+ * been assembled in a structured array and may be used for altering that
+ * listing.
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.api.php]: Line lengths should be as close to 80 characters as possible (https://docs.backdropcms.org/doc-standards#general):

```suggestion
 * This hook is called after the node listing for a taxonomy term page has been
 * assembled in a structured array and may be used for altering that listing.
```

File: core/modules/taxonomy/taxonomy.api.php [hunk 5]
```diff
@@ -272,6 +272,24 @@ function hook_taxonomy_term_view_alter(&$build) {
...
+ *
+ * @param $build
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.api.php]: Need to add the data type indicator here (https://docs.backdropcms.org/doc-standards#param):

```suggestion
 * @param array $build
```

File: core/modules/taxonomy/taxonomy.api.php [hunk 6]
```diff
@@ -272,6 +272,24 @@ function hook_taxonomy_term_view_alter(&$build) {
...
+ *   A renderable array representing the taxonomy term content.
+ *
+ * @see taxonomy_term_page()
+ */
+function hook_taxonomy_term_page_alter(&$build) {
+  // Remove the list of nodes with this term (build).
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.api.php]: Unsure what the purpose of `(build)` here is...?

File: core/modules/taxonomy/taxonomy.api.php [hunk 7]
```diff
@@ -272,6 +272,24 @@ function hook_taxonomy_term_view_alter(&$build) {
...
+  unset($build['nodes']);
+  // Remove the pager for the list of nodes.
+  unset($build['pager']);
+}
```
Comment (ghost) [inline core/modules/taxonomy/taxonomy.api.php:291]: Need blank line after this:
```suggestion
}

```

**Review by ghost — CHANGES_REQUESTED** [review]
Few documentation/coding standards needed here...

File: core/modules/taxonomy/taxonomy.api.php (hunk 6 above)
Comment (remote007) [reply core/modules/taxonomy/taxonomy.api.php]: It is the variable list parameter passed to the function.

Comment (ghost) [reply core/modules/taxonomy/taxonomy.api.php]: I mean in the comment: "Remove the list of nodes with this term (build)."

**Review by ghost — APPROVED** [review]

**Comment by quicksketch** [comment]
Merged into 1.x and 1.21.x. Thanks @remote007!

**Pull request merged**